2_process/src/raster_in_polygon_weighted_mean.R

```diff
@@ -0,0 +1,47 @@
+raster_in_polygon_weighted_mean <- function(raster,
+                                            nhd_polygon_layer,
+                                            weighted_mean_col_name = 'dtb_weighted_mean',
```

Comment:
The function name implies that this function could be used more generally for any use case where we want to extract raster data using a polygon as a mask. _If_ that's the case, I'd suggest that we not use `dtb_weighted_mean` as a default value for the argument `weighted_mean_col_name` and instead just make the user choose each time 🙂